src/backend/parser/parse_type.c

```diff
@@ -361,6 +362,10 @@ typenameTypeMod(ParseState *pstate, const TypeName *typeName, Type typ)
 
 	typmodin = ((Form_pg_type) GETSTRUCT(typ))->typmodin;
 
+	/* Handle the typmodin for domains like smallmoney/money and UDTs on them. */
+	if (handle_basetype_typmodin_hook)
+		(*handle_basetype_typmodin_hook)(typ, &typmodin);
```

Comment:
According to the comment above, shouldn't we call this handle_domain_typmodin_hook?